convert2rhel/breadcrumbs.py

```diff
@@ -87,8 +87,10 @@ def finish_collection(self, success=False):
             self._set_target_os()
 
         self._set_ended()
+
         self._save_migration_results()
-        self._save_rhsm_facts()
+        if self.activity_started != "null" and "CONVERT2RHEL_DISABLE_TELEMETRY" not in os.environ:
```

Comment:
```suggestion
        if "CONVERT2RHEL_DISABLE_TELEMETRY" not in os.environ:
```
The condition of activity_started IMHO doesn't need to be here. It was not part of the Jira issues to not upload the facts when activity_started is empty. I think we can save the facts file with or without the `activity_started` timestamp.

At this point we have at least the `activity_ended` stored (see `self._set_ended()` two lines above). Which is quite weird - to have the `ended` timestamp but not the `started` timestamp. For that reason I'd simply change the class init to have:
`self.activity_started = self._get_formated_time()`
instead of
`self.activity_started = "null"`

By the way the spelling of `_get_formated_time` is wrong, it should be `_get_formatted_time` 😜

Reply:
@bocekm The condition is there for not saving the `rhsm_facts` if user declined the collection. I've done in this way because in that moment we don't have set these data - the data are set after this step. If we won't use the time default value I can choose something different what is collected during `collect_early_data`.

@danmyway noticed that (without this part of condition) if user declines the data collection the message about saving the `rhsm-facts` is printed.

Reply:
The way you implemented it is quite confusing. Not only because you've picked a random breadcrumbs piece of many to check if it's set, but there's not even a comment why have you done so and what's the purpose of the `if self.activity_started != "null"` condition.

I'd remove this `if self.activity_started != "null"` part of the condition and let the breadcrumbs facts file be created even with null values. We can discuss next steps on our 1:1.

Reply:
During the discussion with @hosekadam we've concluded that we should not create the facts file before the user acknoledges the telemetry. For that reason @hosekadam will add a flag to the Breadcrumbs class indicating whether the user has already acknowledged it and then use this flag in this condition.

Reply:
@danmyway @bocekm Flag added, tested.